File: gscientist/server/services/project_service.py
from gscientist.project_manager import ProjectManager

class ProjectService:
    def __init__(self, base_path=None, config_dir=None):
        self.pm = ProjectManager(base_path=base_path, config_dir=config_dir)

    def list_projects(self):
        return self.pm.list_projects()

    def get_project(self, name):
        return self.pm.get_project(name)

    def create_project(self, name, description=""):
        return self.pm.create_project(name, description)

    def rename_project(self, name, new_name):
        return self.pm.rename_project(name, new_name)

    def delete_project(self, name):
        return self.pm.delete_project(name)

File: ui/frontend/js/services/projectService.js
// ui/frontend/js/services/projectService.js
class ProjectService {
    constructor(baseUrl = 'http://localhost:8000') {
        this.baseUrl = baseUrl;
    }

    async listProjects() {
        const res = await fetch(`${this.baseUrl}/projects/`);
        if (!res.ok) throw new Error('Failed to fetch projects');
        return (await res.json()).projects;
    }

    async createProject(name, description = '') {
        const res = await fetch(`${this.baseUrl}/projects/`, {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify({ name, description })
        });
        if (!res.ok) throw new Error('Failed to create project');
        return await res.json();
    }

    async renameProject(name, newName) {
        const res = await fetch(`${this.baseUrl}/projects/${encodeURIComponent(name)}/rename`, {
            method: 'PUT',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify({ new_name: newName })
        });
        if (!res.ok) throw new Error('Failed to rename project');
        return await res.json();
    }

    async deleteProject(name) {
        const res = await fetch(`${this.baseUrl}/projects/${encodeURIComponent(name)}`, {
            method: 'DELETE'
        });
        if (!res.ok) throw new Error('Failed to delete project');
        return await res.json();
    }
}

export default new ProjectService();
